dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,695 @@
+/*
+    DSSCollection contains collection group & completion functionality for DSS.
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import NonFungibleToken from "./NonFungibleToken.cdc"
+import MetadataViews from 0xMETADATAVIEWSADDRESS
+
+// The DSSCollection contract
+//
+pub contract DSSCollection: NonFungibleToken {
+
+    // Contract Events
+    //
+    pub event ContractInitialized()
+
+    // NFT Collection Events
+    //
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+
+    // Events
+    //
+    pub event CollectionGroupCreated(
+        id: UInt64,
+        name: String,
+        description: String,
+        typeName: String,
+        startTime: UFix64?,
+        endTime: UFix64?,
+        timeBound: Bool
+    )
+    pub event CollectionGroupClosed(id: UInt64)
+    pub event ItemCreated(
+        id: UInt64,
+        itemID: UInt64,
+        points: UInt64,
+        itemType: String
+    )
+    pub event SlotCreated(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        logicalOperator: String,
+        typeName: String
+    )
+    pub event ItemAddedToSlot(
+        id: UInt64,
+        slotID: UInt64,
+        collectionGroupID: UInt64
+    )
+    pub event CollectionNFTMinted(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        serialNumber: UInt64,
+        completedBy: String,
+        completionDate: UFix64,
+        level: UInt8
+    )
+    pub event CollectionNFTBurned(id: UInt64)
+
+    // Named Paths
+    //
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    // Entity Counts
+    //
+    pub var totalSupply:    UInt64
+    pub var nextItemID:     UInt64
+
+    // Lists in contract
+    //
+    access(self) let collectionGroupByID: @{UInt64: CollectionGroup}
+    access(self) let slotByID: @{UInt64: Slot}
+    access(self) let itemByID: {UInt64: Item}
+
+    // A public struct to access Item data
+    //
+    pub struct Item {
+        pub let id: UInt64 // unique uuid of resource
+        pub let itemID: UInt64 // the id of the edition, tier, play
+        pub let points: UInt64 // points for item
+        pub let itemType: String // (edition.id, edition.tier, play.id)
+
+        init (
+            id: UInt64,
+            itemID: UInt64,
+            points: UInt64,
+            itemType: String
+        ) {
+            self.id = id
+            self.itemID = itemID
+            self.points = points
+            self.itemType = itemType
+
+            DSSCollection.nextItemID = self.id + 1 as UInt64
```

Comment:
Nit: the id should be set to DSSCollection.nextItemID in the same place it is incremented. So instead of passing an id to init, we do `self.id = DSSCollection.nextItemID`

Reply:
+1

Reply:
Yes, totally agree. I have fixed this.